addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/internal/discovery/BleboxScanner.java

```
@@ -0,0 +1,239 @@
+            String hostAddress = inet4.get(0).getHostAddress();
+            subnetUtils = new SubnetUtils(hostAddress + "/" + "24");
+            subnetInfo = subnetUtils.getInfo();
+            lowIP = convertIPToNumber(subnetInfo.getLowAddress());
+            highIP = convertIPToNumber(subnetInfo.getHighAddress());
+        } catch (IllegalArgumentException e) {
+            logger.error("discoverBridge(): Illegal Argument Exception - {}", e.toString());
+            return;
+        } catch (Exception e) {
+            logger.error("discoverBridge(): Error - Unable to get Subnet Information! {}", e.toString());
+            return;
+        }
+
+        logger.debug("   Local IP Address: {} - {}", subnetInfo.getAddress(),
+                convertIPToNumber(subnetInfo.getAddress()));
+        logger.debug("   Subnet:           {} - {}", subnetInfo.getNetworkAddress(),
+                convertIPToNumber(subnetInfo.getNetworkAddress()));
+        logger.debug("   Network Prefix:   {}", subnetInfo.getCidrSignature().split("/")[1]);
+        logger.debug("   Network Mask:     {}", subnetInfo.getNetmask());
+        logger.debug("   Low IP:           {}", convertNumberToIP(lowIP));
+        logger.debug("   High IP:          {}", convertNumberToIP(highIP));
+
+        ExecutorService threadpool = Executors.newFixedThreadPool(50);
+        Async async = Async.newInstance().use(threadpool);
+
+        for (long ip = lowIP; ip <= highIP; ip++) {
+            try {
+                final String ipAddress = convertNumberToIP(ip);
+
+                Gson gson = new Gson();
+
+                URIBuilder builder = new URIBuilder();
+                builder.setScheme("http").setHost(ipAddress).setPath("/api/device/state");
+                URI requestURL = null;
+                try {
+                    requestURL = builder.build();
+                } catch (URISyntaxException use) {
+                }
```

Empty catch block hides problems.